cmake/Modules/Packages/GPU.cmake

```diff
@@ -407,6 +417,10 @@ elseif(GPU_API STREQUAL "HIP")
     target_compile_definitions(gpu PRIVATE -DMPI_GERYON -DUCL_NO_EXIT)
   endif()
   target_link_libraries(gpu PRIVATE hip::host)
+  if(BUILD_OMP)
```

Comment:
Wouldn't it be enough to add this to the end of the file once?

Reply:
I have thought about that, but didn't like the disconnect. It is more obvious, if this command is where the other targets are added.

Reply:
So instead we duplicate code that we now have to maintain in multiple places?

Reply:
Most of this file is an exercise in duplication...

Reply:
"triplication" to be more accurate. The way I see it is the following: I already *know* that I have to check everything 3 times (CUDA, OpenCL, HIP), by moving the statement outside the sections, I have to remember to look into a fourth place.

Reply:
I don't see that as a reason to add more cruft.